### ISSUE TITLE
Tag inference is not working properly.

### ISSUE THREAD

**Issue by Luffbee** [danburkert/prost#205]
The following code will panic with "assertion failed: tag >= MIN_TAG && tag <= MAX_TAG" because `tag` is 0u32 (`MIN_TAG` is 1u32). This is caused by the initial value of `next_tag` is 0u32.

The `message_encoding` test for tag inference is passed because the first field is a `bool`, and it only checks the `default()`, which makes the field be a zero value and be skipped during encoding.

``` rust
#[derive(Clone, PartialEq, Message)]
pub struct TagsInferred {
    #[prost(bool)]
    pub one: bool,
    #[prost(int32, optional)]
    pub two: Option<i32>,
    #[prost(float, repeated)]
    pub three: Vec<f32>,
}

fn main() {
    let x = TagsInferred {
        one: true,
        two: None,
        three: vec![0.5],
    };
    let mut buf = Vec::new();
    x.encode(&mut buf).unwrap();
    println!("{:?}", buf);
}
```

**Comment by Luffbee** [comment]
https://github.com/danburkert/prost/blob/1944c27c3029d01ff216e7b126d846b6cf8c7d77/prost-derive/src/lib.rs#L55

**Comment by Luffbee** [comment]
https://github.com/danburkert/prost/blob/1944c27c3029d01ff216e7b126d846b6cf8c7d77/tests/src/message_encoding.rs#L192-L218

**Comment by Luffbee** [comment]
Fixed by PR #206
